js/components/addressBook/addressBook_controller.js

```diff
@@ -103,16 +104,26 @@ angular.module('contactsApp')
 
 			// Now filter out all sharees that are already shared with
 			for (i = 0; i < userSharesLength; i++) {
-				var share = userShares[i];
-				usersLength = users.length;
-				for (j = 0; j < usersLength; j++) {
-					if (users[j].value.shareWith === share.id) {
+				var shareUser = userShares[i];
+				for (j = 0; j < users.length; j++) {
+					if (users[j].value.shareWith === shareUser.id) {
```

Comment:
How about using `Array.filter` instead of the `for splice` construct to improve readability?

Reply:
We could indeed, but we need to create the filter function anyway, so it would basically be the same :)

Reply:
Also: basic loop seems more performant: https://jsperf.com/javascript-array-filter-versus-loop

Reply:
Splice seems to defeat the performance improvement:
https://jsperf.com/splice-vs-filter

But let's keep it as it is for now.
We don't need to overengineer this.

Reply:
We could use an auto test of performance? Is such thing possible?